common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -75,13 +83,23 @@ public MLAgent(String name,
         this.createdTime = createdTime;
         this.lastUpdateTime = lastUpdateTime;
         this.appType = appType;
+        this.isHidden = isHidden;
         validate();
     }
 
     private void validate() {
         if (name == null) {
-            throw new IllegalArgumentException("agent name is null");
+            throw new IllegalArgumentException("Agent name can't be null");
+        }
+        if (type == null) {
+            throw new IllegalArgumentException("Agent type can't be null");
+        } else {
+            validateMLAgentType(type);
+        }
```

Comment:
Nit: can move null check as part of validateMLAgentType

Reply:
done